LanguageFeatures/Patterns/variable_A02_t01.dart

```diff
@@ -47,7 +47,26 @@ String testRecord(Record r, [bool doTypeTest = false]) {
   }
 }
 
-String testList(List l, [bool doTypeTest = false]) {
+String testRecord2(Record r) {
+  return switch (r) {
+    case (var a, final b) => "($a, $b)";
+    default => "default";
+  };
+}
+
+String testRecord3(Record r, [bool doTypeTest = false]) {
+  if (r case (var a, final b)) {
+    if (doTypeTest) {
+      a.expectStaticType<Exactly<int>>();
```

Comment:
I don't understand this one. Isn't `Record` the type from 'dart:core' which is a supertype of all record types? In that case I don't see any elements of the code where `a` would be associated with the type `int` during static analysis.

I would expect that the fields of a receiver of type `Record` would have type `dynamic` or `Object?`, cf. https://github.com/dart-lang/language/issues/2665.

Similarly for line 61.

Reply:
Actually (and luckily ;-), `Record` is much more static than `dynamic`: It simply does not have any members other than the ones in `Object`.

This means that there is no help from a context type when it comes to the inferred declared type of `a` and `b`.

I think this area is somewhat underspecified, but let's settle for `Object?` based on [this sentence](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=the%20required%20type%20of%20p%20is%20a%20record%20type%20with%20the%20same%20shape%20as%20p%20and%20object%3F%20for%20all%20fields.).